File: python/hackerrank/equality-in-a-array.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the equalizeArray function below.
def equalizeArray(arr):
    counts = {}
    max_count = 0
    for i in arr:
        if i in counts:
            counts[i] += 1
        else:
            counts[i] = 1
        max_count = max(max_count, counts[i])
    return len(arr) - max_count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    result = equalizeArray(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: ruby/hackerrank/equality-in-a-array.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the equalizeArray function below.
def equalizeArray(arr)
    counts = {}
    max_count = 0
    arr.each do |i|
        counts[i] = if counts[i] then counts[i] + 1 else 1 end
        max_count = [max_count, counts[i]].max
    end
    return arr.size - max_count

end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

n = gets.to_i

arr = gets.rstrip.split(' ').map(&:to_i)

result = equalizeArray arr

fptr.write result
fptr.write "\n"

fptr.close()
